projects/core/src/features-config/utils/use-feature-styles.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DestroyRef, inject } from '@angular/core';
+import { FeatureStylesService } from '../directives/feature-styles.service';
+
+export function useFeatureStyles(feature: string): void {
+  const featureStylesService = inject(FeatureStylesService);
+  const destroyRef = inject(DestroyRef);
+
+  featureStylesService.registerUsage(feature);
+  destroyRef.onDestroy(() => featureStylesService.unregisterUsage(feature));
+}
```

Comment:
```suggestion
export function useFeatureStyles(feature: string): void {
  assertInInjectionContext(useFeatureStyles);

  const featureStylesService = inject(FeatureStylesService);
  const destroyRef = inject(DestroyRef);

  featureStylesService.registerUsage(feature);
  destroyRef.onDestroy(() => featureStylesService.unregisterUsage(feature));
}
```

Thanks to `assertInInjectionContext`Angular will throw an error using function's name if the function is used outside the injection context:
`Error: NG0203: useFeatureStyles() can only be used within an injection context such as a constructor, a factory function, a field initializer, or a function used with `runInInjectionContext`.`

Reply:
good to know. thanks!